homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -178,17 +178,18 @@ def update(self):
             avoid_ferries = self.config_entry.options[CONF_AVOID_FERRIES]
             units = self.config_entry.options[CONF_UNITS]
 
+            routes = {}
             try:
-                params = WazeRouteCalculator(
-                    self.origin,
-                    self.destination,
-                    self.region,
-                    vehicle_type,
-                    avoid_toll_roads,
-                    avoid_subscription_roads,
-                    avoid_ferries,
-                )
-                routes = params.calc_all_routes_info(real_time=realtime)
+                async with WazeRouteCalculator(
```

Comment:
Can we store the client as an instance attribute so we don't need to create a new client every update?

Reply:
Yes this is far more efficient. Wanted to keep the changes minimal and this usually gets called every 5 minutes.

Can change it though.

Reply:
This would involve a change in pywaze. Will take a bit longer

Reply:
done